plugins/ipam/host-local/main.go

```diff
@@ -26,49 +28,99 @@ func main() {
 }
 
 func cmdAdd(args *skel.CmdArgs) error {
-	ipamConf, err := LoadIPAMConfig(args.StdinData, args.Args)
+	ipamConf, ipamConf6, err := LoadIPAMConfig(args.StdinData, args.Args)
 	if err != nil {
 		return err
 	}
 
-	store, err := disk.New(ipamConf.Name)
-	if err != nil {
-		return err
+	r := &types.Result{
+		IP4: nil,
+		IP6: nil,
 	}
-	defer store.Close()
 
-	allocator, err := NewIPAllocator(ipamConf, store)
-	if err != nil {
-		return err
-	}
+	if ipamConf != nil {
+		store, err := disk.New(ipamConf.Name)
+		if err != nil {
+			return err
+		}
+		defer store.Close()
 
-	ipConf, err := allocator.Get(args.ContainerID)
-	if err != nil {
-		return err
+		allocator, err := NewIPAllocator(ipamConf, store)
+		if err != nil {
+			return err
+		}
+
+		ipConf, err := allocator.Get(args.ContainerID)
+		if err != nil {
+			return err
+		}
+		r.IP4 = ipConf
 	}
 
-	r := &types.Result{
-		IP4: ipConf,
+	if ipamConf6 != nil {
```

Comment:
Thanks for pointing that out. I was trying this out
```
        var allocator IPAllocator
        if ipamConf != nil {
                store, err := disk.New(ipamConf.Name)
                if err != nil {
                        return err
                }
                defer store.Close()

                allocator, err := NewIPAllocator(ipamConf, store)
                if err != nil {
                        return err
                }

                ipConf, err := allocator.Get(args.ContainerID)
                if err != nil {
                        return err
                }
                r.IP4 = ipConf
        }

        if ipamConf6 != nil {
                store, err := disk.New(ipamConf6.Name)
                if err != nil {
                        allocator.Release(args.ContainerID)
                        return err
                }
                defer store.Close()

                allocator6, err := NewIPAllocator(ipamConf6, store)
                if err != nil {
                        allocator.Release(args.ContainerID)
                        return err
                }
                ipConf6, err := allocator6.Get(args.ContainerID)
                if err != nil {
                        allocator.Release(args.ContainerID)
                        return err
                }
                r.IP6 = ipConf6
        }
```
Is there a better way to do this?